IA/IA_upload.py

```diff
@@ -49,6 +53,31 @@ def mp_from_ids(mp_id: str, mp_keyname: str, bucket: str) -> MultiPartUpload:
     await asyncio.gather(*tasks)
 
 
+def upload_metadata(bucket_name: str, guid: str, directory: str):
+    session_data = {'s3': {'access': IA_ACCESS_KEY, 'secret': IA_SECRET_KEY}}
+    session = internetarchive.get_session(config=session_data)
+
+    node_path = os.path.join(HERE, directory, 'data', guid, 'node', '{}.json'.format(guid))
+    with open(node_path, 'r') as f:
+        node_json = json.loads(f.read())['attributes']
+
+    xml_path = os.path.join(HERE, directory, 'data', 'datacite.xml')
+    xml_data = minidom.parse(xml_path)
+    creators = xml_data.getElementsByTagName('creatorName')
+
+    item = session.get_item(bucket_name)
+    item.modify_metadata(dict(
+        title=node_json['title'],
+        description=node_json['description'],
+        creator=creators[0].firstChild.data,
+        date=node_json['date_created'],
+        doi=node_json['article_doi'],
```

Comment:
Hi! I am Internet Archive staff helping with this project, hope it is ok for me to comment here directly.

Can't remember if we specified this correctly in earlier docs, but the correct way to include a DOI (identifier) in IA metadata is under the `external-identifier` field encoded as a URN.

Eg:
```
metadata = dict()
metadata['external-identifier'] = "urn:doi:{}".format(node_json['article_doi'])
```
This is assuming that the DOI is associated with the work being uploaded into the current item (OSF registration). If the DOI is associated with the final publication (paper) that this registration is only associated with, you would use the `related-external-id` field.

IA docs: https://archive.org/services/docs/api/metadata-schema/index.html#external-identifier